---
title: "Targeted Biomarker Discovery"
subtitle: "Biostatistics Admit Day"
author: "[Nima Hejazi](http://nimahejazi.org)"
date: "20 March 2017"
output:
  xaringan::moon_reader:
    lib_dir: libs
    chakra: libs/remark-latest.min.js
    css: ["default", "custom.css"]
    nature:
      highlightStyle: zenburn
      highlightLines: true
---

```{r knitr_setup, echo=FALSE, message=FALSE}
knitr::opts_chunk$set(fig.width = 7, fig.height = 4.5,
					  dpi = 300, fig.cap = "", fig.align = "center")
showtext::showtext.opts(dpi = 300)
```

# How to access these slides
## (and associated code)

### Via git:

```{bash, eval=FALSE}
git clone https://github.com/nhejazi/talk_admitday.git
```

### View online:
https://goo.gl/HZzosu

???

- This talk will focus on challenges in analyzing high-dimensional biological
data, with a brief introduction to common problems.

- We will look at how standard approaches from statistical causal inference
(esp., _Targeted Learning_) can be extended to this class of data.

---
class: inverse, center, middle

# Statistics in High-Dimensional Biology

---

# Biological Sequencing Data

- lots of data is constantly collected at the sequencing level...

- experiments in health sciences, epidemiology...

--

- Popular biotechnology: (1) Microarrays, (2) RNA-Seq, (3) Single-Cell RNA-Seq

- Querying different genomic processes: (1) DNA/RNA expression, (2) Protein
expression, (3) Epigenetics: DNA Methylation

???

- There are numerous challenges in analyzing such data sets.

- High-dimensional? $n \in (5, 200)$, $g \in (10000, 850000)$

- one more talking point...

---

# Central Dogma of Molecular Biology

- image here

???

- Why do we care about DNA, RNA, etc.?

- talking point...

- talking point...

---

# Statistical Issues in Genomics

Standard data matrix  vs. Genomic data matrix:

$\begin{bmatrix} x_{11} & x_{12} & x_{13} & \dots & x_{1g} \\ x_{21} & x_{22} & x_{23} & \dots & x_{2g} \\ \vdots & \vdots & \vdots & \ddots & \vdots \\ x_{n1} & x_{n2} & x_{n3} & \dots & x_{ng} \end{bmatrix} \xrightarrow[]{\text{transpose}} \begin{bmatrix} x_{11} & x_{12} & x_{13} & \dots & x_{1n} \\ x_{21} & x_{22} & x_{23} & \dots & x_{2n} \\ \vdots & \vdots & \vdots & \ddots & \vdots \\ x_{g1} & x_{g2} & x_{g3} & \dots & x_{gn} \end{bmatrix}$

__n.b.__, $n << g$

--

- standard practice: _subjects in rows, genes in columns_

- genomics standards: _genes in rows, subjects in columns_

--

- Wait, why did we transpose this matrix weirdly?

- Why this aweful convention? 

--

- Thanks, Microsoft (Excel)...

???

- talking point...

- talking point...

- talking point...

---

# Targeted Learning

- A: exposure/treatment (e.g., benzene)

- Y: outcome of interest (e.g., gene expression)

- W: baseline variables (e.g., age, sex, SES)

--

- __goal:__ estimate the effect of a treatment (A) on an outcome (Y) while
controlling for baseline covariates (W)

- causal inference and variable importance

--

- target parameters like the "average treatment effect" (ATE):

$$\Psi(P_n^*) = E[E[Y \mid A = 1, W] - E[Y \mid A = 0, W]]$$

???

- talking points

- talking points

- talking points

---

# R package: `biotmle`

```{r, eval=FALSE}
library(devtools)
devtools::install_github("nhejazi/biotmle")
```

- what does this package do?

- who cares?

???

- talking point

- talking point

---
class: inverse, middle, center

# Data Analysis with `biotmle`

---

# Explore the data set

```{r}
library(dplyr)
library(biotmle)
data(illuminaData)
```

```{r, echo=FALSE}
"%ni%" = Negate("%in%")
subjIDs <- illuminaData$id
```

- load the data set

- ...

???

- This isn't too important, just click through

---

# Baseline covariates (W)

```{r}
# W - age, sex, smoking
W <- illuminaData %>%
  dplyr::select(which(colnames(.) %in% c("age", "sex", "smoking"))) %>%
  dplyr::mutate(
    age = as.numeric((age > quantile(age, 0.25))),
    sex = I(sex),
    smoking = I(smoking)
  )
head(W)
```

???

- talking point

- talking point

---

# Exposure of interest (A)

```{r}
# A - benzene exposure (discretized)
A <- illuminaData %>%
  dplyr::select(which(colnames(.) %in% c("benzene")))
A <- A[, 1]
unique(A)
table(A)
```

???

- talking

- talking

---

# Outcome of interest (Y)

```{r}
# Y - genes
Y <- illuminaData %>%
  dplyr::select(which(colnames(.) %ni% c("age", "sex", "smoking", "benzene",
                                         "id")))
geneIDs <- colnames(Y)
dim(Y)
head(Y[, 1:7])
```

???

- expression measures

- look at that dimensionality!

---

# Identifying biomarkers

We can use the package to identify potential biomarkers:

```{r eval=FALSE}
biomarkerTMLEout <- biomarkertmle(Y = Y, # biomarkers
                                  W = W, # baseline covariates
                                  A = A, # exposure (benzene)
                                  type = "exposure",
                                  parallel = TRUE,
                                  family = "gaussian",
                                  g_lib = c("SL.glmnet", "SL.randomForest",
                                            "SL.polymars", "SL.mean"),
                                  Q_lib = c("SL.glmnet", "SL.randomForest",
                                            "SL.nnet", "SL.mean")
                                 )
```

```{r, echo=FALSE}
data(biomarkertmleOut)
```

```{r}
design <- as.data.frame(cbind(rep(1, nrow(Y)),
                              as.numeric(A == max(unique(A)))))
colnames(design) <- c("intercept", "Tx")
limmaTMLEout <- limmatmle(biotmle = biomarkerTMLEout, IDs = NULL,
                          designMat = design)
```

???

- The procedure is quite resource-intensive as it evaluates the association of
each potential biomarker (over $20,000$) with the exposure of interest A

- ...while accounting for potential confounding based on the covariates included
in W

---

class: inverse, middle, center

# Visualizing results (from `biotmle`)

---

##### Unadjusted results from our test

```{r, echo=FALSE}
plot(x = limmaTMLEout, type = "pvals_raw")
```

---

##### Adjusted results from our test

```{r, echo=FALSE}
plot(x = limmaTMLEout, type = "pvals_adj")
```

???

- multiple testing

---

##### Visualize results I

```{r, echo=FALSE}
volcano_biotmle(biotmle = limmaTMLEout)
```

???

- volcano plots are pretty standard

---

##### Visualize results II

```{r, echo=FALSE}
heatmap_biotmle(x = limmaTMLEout, left.label.text.size = 2, designMat = design,
                FDRcutoff = 0.05, top = 25)
```

---

class: center, middle

# Thanks!

Slides created via the R package [**xaringan**](https://github.com/yihui/xaringan).

Powered by [remark.js](https://remarkjs.com),
[**knitr**](http://yihui.name/knitr), and
[R Markdown](https://rmarkdown.rstudio.com).
